### PR TITLE
Use Go types in pvl

### PR DESCRIPTION
**Pull request by mlsteele**
This is in line after https://github.com/keybase/client/pull/4371

Switched to using Go types in the interpreter. This marks the switch to instructions like `{"assert_regex_match": {"pattern": "^foo$" } }`. Named registers are yet to come.

r? @oconnor663

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 796b9101b755...b55b94653b26 introduced some visual changes on linux:
 * Changed: **New_Proof_PGP_generating-Generating_PGP** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1474556403050-796b9101b755-b55b94653b26-linux/New_Proof_PGP_generating-Generating_PGP-796b9101b755.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1474556403050-796b9101b755-b55b94653b26-linux/New_Proof_PGP_generating-Generating_PGP-b55b94653b26.png)  
   New_Proof_PGP_generating-Generating_PGP comparison

File: go/pvl/interp.go [hunk 1]
```diff
@@ -510,62 +448,68 @@ func runScript(g proofContextExt, script *jsonw.Wrapper, startstate scriptState)
...
+	case ins.SelectorCSS != nil:
+		newState, stepErr = stepSelectorCSS(g, *ins.SelectorCSS, state)
+		customErrSpec = ins.SelectorCSS.Error
+	case ins.TransformURL != nil:
+		newState, stepErr = stepTransformURL(g, *ins.TransformURL, state)
+		customErrSpec = ins.TransformURL.Error
+	default:
+		newState = state
+		stepErr = libkb.NewProofError(keybase1.ProofStatus_INVALID_PVL,
+			"Invalid instruction")
```
Comment (oconnor663) [inline go/pvl/interp.go]: Can we say what the instruction is here?

**Comment by mlsteele** [comment]
> Can we say what the instruction is here?

Yes! done.

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 9363ab6206e5...6eddbca7dfce introduced some visual changes on linux:
 * Changed: **unlock_folders-Single** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475001849896-9363ab6206e5-6eddbca7dfce-linux/unlock_folders-Single-9363ab6206e5.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475001849896-9363ab6206e5-6eddbca7dfce-linux/unlock_folders-Single-6eddbca7dfce.png)  
   unlock_folders-Single comparison
 * Changed: **unlock_folders-No_paperkeys** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475001849896-9363ab6206e5-6eddbca7dfce-linux/unlock_folders-No_paperkeys-9363ab6206e5.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475001849896-9363ab6206e5-6eddbca7dfce-linux/unlock_folders-No_paperkeys-6eddbca7dfce.png)  
   unlock_folders-No_paperkeys comparison

File: go/pvl/interp.go [hunk 2]
```diff
@@ -510,62 +448,68 @@ func runScript(g proofContextExt, script *jsonw.Wrapper, startstate scriptState)
...
-				stepErr = customErr
-				debugWithState(g, state, "Replacing error with custom error")
-				debugWithStateError(g, state, customErr)
-			}
+		// Replace error with custom error.
+		customErr, swap := replaceCustomError(g, state, customErrSpec, stepErr)
+		if swap {
+			stepErr = customErr
+			debugWithState(g, state, "Replacing error with custom error")
+			debugWithStateError(g, state, customErr)
```
Comment (oconnor663) [inline go/pvl/interp.go]: Would it make more sense to unconditionally assign the return value to `stepErr`, and to move the logging down into `replaceCustomError`? Then you might not need the bool.

Comment (mlsteele) [inline go/pvl/interp.go]: Sounds good, done

**Pull request merged**